### PR TITLE
[FIX] Email verification on update not working for users in secondary userstore

### PR DESCRIPTION
**Pull request by somindatommy**
Resolves https://github.com/wso2/product-is/issues/7710

## Proposed changes in this pull request
Removed domain name getting appended to username.
Improved debug logs.

Support Fix: https://github.com/wso2-support/identity-governance/pull/182

### CODE REVIEW

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java [hunk 1]
```diff
@@ -364,20 +364,21 @@ private void preSetUserClaimsOnEmailUpdate(Map<String, String> claims, UserStore
         if (StringUtils.isNotBlank(emailAddress)) {
 
             String existingEmail;
-            String username = IdentityUtil.addDomainToName(user.getUserName(), user.getUserStoreDomain());
+            String username = user.getUserName();
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java]: Don't we need to append userstore domain here?
does user.getUserName() gives name with userstore domain

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java [hunk 2]
```diff
@@ -364,20 +364,21 @@ private void preSetUserClaimsOnEmailUpdate(Map<String, String> claims, UserStore
...
             try {
                 existingEmail = userStoreManager.getUserClaimValue(username,
                         IdentityRecoveryConstants.EMAIL_ADDRESS_CLAIM, null);
             } catch (UserStoreException e) {
-                throw new IdentityEventException("Error occurred while retrieving existing email address for user: "
-                        + username, e);
+                String error = String.format("Error occurred while retrieving existing email address for user: %s in " +
+                        "domain : %s", user, user.getTenantDomain());
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java]: user will be an object right, we need to print the username here.

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java [hunk 3]
```diff
@@ -421,8 +422,7 @@ private String getPendingVerificationEmailValue(UserStoreManager userStoreManage
 
         Map<String, String> verificationPendingEmailClaimMap;
         try {
-            verificationPendingEmailClaimMap = userStoreManager.getUserClaimValues(IdentityUtil.addDomainToName(
-                    user.getUserName(), user.getUserStoreDomain()), new String[]{
+            verificationPendingEmailClaimMap = userStoreManager.getUserClaimValues(user.getUserName(), new String[]{
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java]: Check whether username contains the domain or whether its needed?

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java (hunk 1 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java]: Appending the userstore name was the error.

user.getUserName() does not contain the userstore domain

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java (hunk 3 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java]: `user.getUserName()` does not contain domain. To get the domain `user.getTenantDomain` method needs to be used.

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java (hunk 2 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java]: Fixed: 105be10d6a8db9038911f8f4bdaf0f16e32fae94

**Pull request merged**